hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -131,7 +132,30 @@ public static DBCheckpoint createOmSnapshotCheckpoint(
       OMMetadataManager omMetadataManager, SnapshotInfo snapshotInfo)
       throws IOException {
     RDBStore store = (RDBStore) omMetadataManager.getStore();
-    return store.getSnapshot(snapshotInfo.getCheckpointDirName());
+
+    final long dbLatestSequenceNumber =
+        ((RDBStore) omMetadataManager.getStore()).getDb()
+            .getLatestSequenceNumber();
+
+    final RocksDBCheckpointDiffer dbCpDiffer =
+        omMetadataManager.getStore().getRocksDBCheckpointDiffer();
+
+    final DBCheckpoint dbCheckpoint = store.getSnapshot(
+        snapshotInfo.getCheckpointDirName());
+
+    // Write snapshot generation (latest sequence number) to compaction log.
+    // This will be used for DAG reconstruction as snapshotGeneration.
+    dbCpDiffer.appendSequenceNumberToCompactionLog(dbLatestSequenceNumber);
+
+    // Set compaction log filename to the latest DB sequence number
+    // right after taking the RocksDB checkpoint for Ozone snapshot.
+    //
+    // Note it doesn't matter if sequence number hasn't increased (even though
+    // it shouldn't happen), since the writer always appends the file.
+    dbCpDiffer.setCompactionLogParentDir(store.getSnapshotsParentDir());
```

Comment:
Does the parent dir get set every time a snapshot is created?

Reply:
No it is not necessary. Thanks. Removed.